molgenis-data-security/src/main/java/org/molgenis/data/security/auth/GroupService.java

```diff
@@ -206,20 +198,22 @@ public void updateMemberRole(final Group group, final User member, final Role ne
 			throw new NotAValidGroupRoleException(newRole, group);
 		}
 
-		RoleMembership roleMembership = findRoleMembership(member, groupRoles);
-
+		final RoleMembership roleMembership = findRoleMembership(member, groupRoles);
 		roleMembershipService.updateMembership(roleMembership, newRole);
 	}
 
-	private RoleMembership findRoleMembership(User member, ArrayList<Role> groupRoles)
+	private UnknownEntityException unknownMembershipForUser(User user)
+	{
+		return new UnknownEntityException(roleMembershipMetadata,
+				roleMembershipMetadata.getAttribute(RoleMembershipMetadata.USER), user.getUsername());
+	}
+
+	private RoleMembership findRoleMembership(User member, List<Role> groupRoles)
 	{
 		Collection<RoleMembership> memberships = roleMembershipService.getMemberships(groupRoles);
 		return memberships.stream()
 						  .filter(m -> m.getUser().getId().equals(member.getId()))
-						  .findFirst()
-						  .orElseThrow(() -> new UnknownEntityException(roleMembershipMetadata,
-								  roleMembershipMetadata.getAttribute(RoleMembershipMetadata.USER),
-								  member.getUsername()));
+						  .findFirst().orElseThrow(() -> unknownMembershipForUser(member));
```

Comment:
imo this does not improve readability, i view the initial check a being more readable